packages/angular-cli/models/webpack-build-styles.ts

```diff
@@ -0,0 +1,129 @@
+import * as webpack from 'webpack';
+import * as path from 'path';
+import {
+  SuppressExtractedTextChunksWebpackPlugin
+} from '../plugins/suppress-entry-chunks-webpack-plugin';
+import { extraEntryParser, getOutputHashFormat } from './webpack-build-utils';
+
+const postcssDiscardComments = require('postcss-discard-comments');
+const autoprefixer = require('autoprefixer');
+const ExtractTextPlugin = require('extract-text-webpack-plugin');
+
+/**
+ * Enumerate loaders and their dependencies from this file to let the dependency validator
+ * know they are used.
+ *
+ * require('raw-loader')
+ * require('style-loader')
+ * require('postcss-loader')
+ * require('css-loader')
+ * require('stylus')
+ * require('stylus-loader')
+ * require('less')
+ * require('less-loader')
+ * require('node-sass')
+ * require('sass-loader')
+ */
+
+export function getWebpackStylesConfig(
+  projectRoot: string,
+  appConfig: any,
+  target: string,
+  sourcemap: boolean,
+  outputHashing: string,
+  extractCss: boolean,
+) {
+
+  const appRoot = path.resolve(projectRoot, appConfig.root);
+  const entryPoints: { [key: string]: string[] } = {};
+  const globalStylePaths: string[] = [];
+  const extraPlugins: any[] = [];
+
+  // discard comments in production
+  const extraPostCssPlugins = target === 'production' ? [postcssDiscardComments] : [];
+
+  // determine hashing format
+  const hashFormat = getOutputHashFormat(outputHashing);
+
+  // use includePaths from appConfig
+  const includePaths: string [] = [];
+
+  if (appConfig.stylePreprocessorOptions
+    && appConfig.stylePreprocessorOptions.includePaths
+    && appConfig.stylePreprocessorOptions.includePaths.length > 0
+  ) {
+    appConfig.stylePreprocessorOptions.includePaths.forEach((includePath: string) =>
+      includePaths.push(path.resolve(appRoot, includePath)));
```

Comment:
Yeah in general anything that's within app config is always relative to `appRoot`. This way it's consistent.